elooplib/eventloop.cpp

```diff
@@ -1,49 +1,33 @@
 #include "eventloop.hpp"
 
-EventLoop::EventLoop(size_t n_clients) : fds(n_clients), max_pollfd_pos(0),
-                                         th_pool(new ThreadPool(3)){};
+EventLoop::EventLoop(size_t n_clients) : max_pollfd_pos(0), fds(n_clients),
+                                         nodes(n_clients, nullptr), th_pool(new ThreadPool(get_nprocs())){};
```

Comment:
You forgot to include a header for get_nprocs

Reply:
i did it in hpp file, sir

Reply:
Ah, I see. It's better to do it include it inside cpp file, because what you include in your headers is seen to everyone else. And this function - is private implantation detail